### PR TITLE
Some smaller fixes

### PR DESCRIPTION
**Pull request by s-scherrer**
- added pyproject.toml (#211)
- removed dependency on deprecated numpy API (#214)
- better calculation of moments in cython metrics
- updated PyScaffold to version 4
- use pytest directly in GH actions
- added pre-commit hooks for running flake8 on diff

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -132,19 +143,19 @@
 # keep_warnings = False
 
 
-# -- Options for HTML output ---------------------------------------------------
+# -- Options for HTML output -------------------------------------------------
 
 # The theme to use for HTML and HTML Help pages.  See the documentation for
 # a list of builtin themes.
-html_theme = 'sphinx_rtd_theme'
+html_theme = "alabaster"
```
Comment (wpreimes) [inline docs/conf.py:150]: I think we use the other template normally, I dont like that one ;)

File: setup.cfg [hunk 2]
```diff
@@ -99,6 +100,13 @@ markers =
     full_framework : marks slow test that use the whole validation framework (deselect with '-m "not full_framework"')
     slow : marks slow tests (deselect with '-m "not slow"')
 testpaths = tests
+# This removes some of the warnings that show up with pytest but are not an issue
+filterwarnings =
```
Comment (wpreimes) [inline setup.cfg:104]: :+1:

File: src/pytesmo/__init__.py [hunk 3]
```diff
@@ -19,4 +44,4 @@
 
 testdata_path = os.path.join(src_path, '..', 'tests', 'test-data')
```
Comment (wpreimes) [inline src/pytesmo/__init__.py:45]: I wonder if there is a better solution, in the tests it is nice to import the testdata path... but maybe this should not be here...

File: src/pytesmo/time_series/filters.pyx [hunk 4]
```diff
@@ -38,25 +38,16 @@ import platform
...
-    bint npy_isnan(double x)
+from cython cimport floating
 
 
 @cython.boundscheck(False)
 @cython.wraparound(False)
 @cython.cdivision(True)
-def exp_filter(np.ndarray[DTYPE_d, ndim=1] in_data,
-               np.ndarray[DTYPE_d, ndim=1] in_jd,
+def exp_filter(floating [:] in_data,
```
Comment (wpreimes) [inline src/pytesmo/time_series/filters.pyx:49]: for the exponential filter we actually have another package (pyswi) which is not yet public... we should make pyswi public (currently on gitlab.geo.tuwien.ac.at) and then remove it from pytesmo. @sebhahn and @Adeaem were working on this

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -54,8 +54,8 @@ jobs:
         shell: bash -l {0}
         run: |
           python setup.py install
-          python setup.py test --addopts "--cache-clear -m 'not full_framework'"
-          python setup.py test --addopts "--cache-clear -m 'full_framework'"
+          pytest --cache-clear -m 'not full_framework'
```
Comment (wpreimes) [inline .github/workflows/ci.yml:57]: btw, this was the fix for avoiding the segfault. no idea where they came from...

File: tests/test_validation_framework/test_adapters.py [hunk 6]
```diff
@@ -12,191 +12,251 @@
...
 from tests.test_validation_framework.test_datasets import TestDataset
-from ascat.read_native.cdr import AscatSsmCdr
 
-import numpy as np
-import numpy.testing as nptest
-import pandas as pd
-from datetime import datetime
+with warnings.catch_warnings():
+    warnings.filterwarnings("ignore")
+    from ascat.read_native.cdr import AscatSsmCdr
```
Comment (wpreimes) [inline tests/test_validation_framework/test_adapters.py:52]: why is the ascat package throwing so many warnings anyway @sebhahn ?

**Review by wpreimes — CHANGES_REQUESTED** [review]
Can you change the docs template please? Otherwise looks fine

File: src/pytesmo/__init__.py (hunk 3 above)
Comment (s-scherrer) [reply src/pytesmo/__init__.py:45]: I think there is a better solution, but I have to look it up

**Comment by wpreimes** [comment]
I wonder if you need to add `coverage` here https://github.com/s-scherrer/pytesmo/blob/916defdc284288b33fa2fb666d98b85f9408f825/setup.cfg#L65

File: .coveragerc [hunk 7]
```diff
@@ -2,12 +2,6 @@
 [run]
 branch = True
 source = pytesmo
-# omit = bad_file.py
-
-[paths]
-source =
-    src/
-    */site-packages/
```
Comment (s-scherrer) [inline .coveragerc:10]: This was causing the issue with coverage, @wpreimes

File: src/pytesmo/time_series/filters.pyx (hunk 4 above)
Comment (s-scherrer) [reply src/pytesmo/time_series/filters.pyx:49]: I only updated the existing code here to not depend on the deprecated numpy API anymore. I would suggest to postpone the removal to another PR, so I can merge this here in soon.

Comment (wpreimes) [reply src/pytesmo/time_series/filters.pyx:49]: sure, we should only remove it when pyswi is public

File: src/pytesmo/__init__.py (hunk 3 above)
Comment (s-scherrer) [reply src/pytesmo/__init__.py:45]: testdata_path is only used in the examples anyways, I changed the examples accordingly and removed it from the package.

**Comment by s-scherrer** [comment]
- added pyproject.toml (#211)
- removed dependency on deprecated numpy API (#214)
- better calculation of moments in cython metrics
- updated PyScaffold to version 4
- use pytest directly in GH actions
- added pre-commit hooks for running flake8 on diff

**Pull request merged**